rust-runtime/aws-smithy-client/src/timeout.rs

```diff
@@ -51,8 +62,163 @@ impl Settings {
     /// Sets the read timeout
     pub fn with_read_timeout(self, read_timeout: Duration) -> Self {
         Self {
-            http_read_timeout: Some(read_timeout),
+            read_timeout: Some(read_timeout),
+            ..self
+        }
+    }
+
+    /// The configured TLS negotiation timeout
+    pub fn with_tls_negotiation_timeout(self, tls_negotiation_timeout: Duration) -> Self {
```

Comment:
we should probably not add these setters until the feature is actually supported